libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationTest.java

```diff
@@ -28,199 +28,251 @@
     navigationWithOptions.removeMilestone(milestone);
+
     assertEquals(1, navigationWithOptions.getMilestones().size());
   }
 
   @Test
   public void removeMilestone_nullRemovesAllMilestones() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder().defaultMilestonesEnabled(false).build();
-    MapboxNavigation navigationWithOptions = new MapboxNavigation(mock(Context.class),
-      ACCESS_TOKEN, options, mock(NavigationTelemetry.class), mock(LocationEngine.class));
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
     navigationWithOptions.addMilestone(new StepMilestone.Builder().build());
     navigationWithOptions.addMilestone(new StepMilestone.Builder().build());
     navigationWithOptions.addMilestone(new StepMilestone.Builder().build());
     navigationWithOptions.addMilestone(new StepMilestone.Builder().build());
-    assertEquals(4, navigationWithOptions.getMilestones().size());
+
     navigationWithOptions.removeMilestone(null);
+
     assertEquals(0, navigationWithOptions.getMilestones().size());
   }
 
   @Test
   public void removeMilestone_correctMilestoneWithIdentifierGetsRemoved() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder().defaultMilestonesEnabled(false).build();
-    MapboxNavigation navigationWithOptions = new MapboxNavigation(mock(Context.class),
-      ACCESS_TOKEN, options, mock(NavigationTelemetry.class), mock(LocationEngine.class));
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
     Milestone milestone = new StepMilestone.Builder().setIdentifier(5678).build();
     navigationWithOptions.addMilestone(milestone);
-    assertEquals(1, navigationWithOptions.getMilestones().size());
+
     navigationWithOptions.removeMilestone(5678);
+
     assertEquals(0, navigationWithOptions.getMilestones().size());
   }
 
   @Test
   public void removeMilestone_noMilestoneWithIdentifierFound() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder().defaultMilestonesEnabled(false).build();
-    MapboxNavigation navigationWithOptions = new MapboxNavigation(mock(Context.class),
-      ACCESS_TOKEN, options, mock(NavigationTelemetry.class), mock(LocationEngine.class));
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
     navigationWithOptions.addMilestone(new StepMilestone.Builder().build());
-    assertEquals(1, navigationWithOptions.getMilestones().size());
+
     navigationWithOptions.removeMilestone(5678);
+
     assertEquals(1, navigationWithOptions.getMilestones().size());
   }
 
   @Test
   public void getLocationEngine_returnsCorrectLocationEngine() throws Exception {
+    MapboxNavigation navigation = buildMapboxNavigation();
     LocationEngine locationEngine = mock(LocationEngine.class);
     LocationEngine locationEngine2 = mock(LocationEngine.class);
```

Comment:
We should avoid adding "numbers" to variable names. Could we refactor-rename adding more context so it'll be easier to read and understand what's going on in the test?